### ISSUE TITLE
[java] UselessParentheses: False negative when on the right-hand side of an assignment statement

### ISSUE THREAD

**Issue by Chordrain** [pmd/pmd#6775]
**PMD Version:** 7.24.0
**Category:** [UselessParentheses](https://docs.pmd-code.org/latest/pmd_rules_java_codestyle.html#uselessparentheses)
**Severity:** Warning (Code Style)

### Summary

`UselessParentheses` reports redundant parentheses around the initializer of a variable declaration (`double value = (a / b);`). The identical redundant parentheses around the right-hand side of a plain assignment statement (`value = (a / b);`) are not reported. The parentheses are equally useless in both positions — removing them changes neither parsing nor operator precedence.

### Reproducer

Before.java (flagged — correct):

```java
public class Before {
    public static void main(String[] argv) {
        int a = Integer.parseInt(argv[0]);
        int b = Integer.parseInt(argv[1]);
        double value = (a / b);
        System.out.println(value);
    }
}
```

After.java (not flagged — missed):

```java
public class After {
    public static void main(String[] argv) {
        int a; a = Integer.parseInt(argv[0]);
        int b; b = Integer.parseInt(argv[1]);
        double value; value = (a / b);
        System.out.println(value);
    }
}
```

### Expected behavior

Both forms should be reported. The parentheses around `a / b` are redundant whether `a / b` is a variable initializer or the right-hand side of an assignment expression.

### Actual behavior

Variable-initializer form → 1 violation:

```
Before.java:5: UselessParentheses: Useless parentheses around `a / b`.
```

Assignment-statement form → 0 violations. The rule does not inspect the RHS of an `AssignmentExpression` for useless parentheses, only `VariableInitializer`.
